lib/scholar/cluster/k_means.ex

```diff
@@ -364,8 +365,8 @@ defmodule Scholar.Cluster.KMeans do
   end
 
   deftransformp validate_weights(weights, num_samples, num_runs) do
-    if is_nil(weights) or
-         (is_list(weights) and length(weights) == num_samples) do
+    if is_nil(weights) or (is_list(weights) and length(weights) == num_samples) or
+         (Nx.is_tensor(weights) and Nx.shape(weights) == {num_samples}) do
```

Comment:
I think you can break this into several blocks:

```
cond do
  is_nil(weights) ->
    1.0
  Nx.is_tensor(weights)  and Nx.shape(weights) == {num_samples} ->
    weights
  is_list(weights) and length(weights) == num_samples ->
    Nx.tensor(weights, type: :f32)
  true ->
    raise "..."
end
```

And then you do the `Nx.as_type(:f32) |> Nx.broadcast({num_runs, num_samples})` inside the defn.